File: front/src/app/core/interfaces/subscription-request.ts
/**
 * Represents a request to subscribe to a theme.
 * 
 * @remarks
 * This interface defines the structure of the request payload needed to subscribe
 * a user to a specific theme. It is typically used when sending a request to an API
 * endpoint to create a new subscription.
 * 
 * @interface
 */
export interface SubscriptionRequest {

    /**
     * The unique identifier of the theme to which the user wants to subscribe.
     * 
     * @remarks
     * This property holds the `theme_id` which represents the specific theme that
     * the user is requesting to subscribe to. It is a numeric value and is required
     * to process the subscription request.
     * 
     * @example
     * ```json
     * {
     *   "theme_id": 123
     * }
     * ```
     */
    theme_id: number;
}

File: back/src/main/java/com/openclassrooms/mddapi/payload/request/SubscriptionRequest.java
package com.openclassrooms.mddapi.payload.request;

import jakarta.validation.constraints.NotNull;

/**
 * Represents a request payload for subscribing to a theme.
 * This class contains the necessary information to create a subscription.
 */
public class SubscriptionRequest {

    /**
     * The ID of the theme to which the user wants to subscribe.
     * This field cannot be null.
     */
    @NotNull
    private Integer theme_id;

    /**
     * Gets the ID of the theme to which the user wants to subscribe.
     * 
     * @return the theme ID
     */
    public Integer getTheme_id() {
        return theme_id;
    }

    /**
     * Sets the ID of the theme to which the user wants to subscribe.
     * 
     * @param theme_id the theme ID
     */
    public void setTheme_id(Integer theme_id) {
        this.theme_id = theme_id;
    }
}
